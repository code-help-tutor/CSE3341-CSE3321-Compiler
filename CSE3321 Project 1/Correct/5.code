
procedure array X , Y , XY 	;

        in X, Y;
		
		while (X === 0) begin
		
			Y = Y * Y;
			X = X - 1;
			
		;
		
		out Y;
		 
		
end  
      
