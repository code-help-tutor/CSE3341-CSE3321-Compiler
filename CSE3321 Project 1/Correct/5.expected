PROCEDURE
ARRAY
ID[X]
COMMA
ID[Y]
COMMA
ID[XY]
SEMICOLON
IN
ID[X]
COMMA
ID[Y]
SEMICOLON
WHILE
LPAREN
ID[X]
EQUAL
EQUAL
EQUAL
CONST[0]
RPAREN
BEGIN
ID[Y]
EQUAL
ID[Y]
MULTIPLY
ID[Y]
SEMICOLON
ID[X]
EQUAL
ID[X]
SUBTRACT
CONST[1]
SEMICOLON
SEMICOLON
OUT
ID[Y]
SEMICOLON
END
